FIPS/fip-0032.md

```diff
@@ -21,290 +21,426 @@ replaces: N/A
-intercepting specific operations and actions, and applying gas explicit gas
-charges on them. Concretely:
-
-- On including a chain message, variable on the length of the message.
-- On returning a value, variable on the length of returned payload.
-- On performing a call, dependent on the type of call (simple value transfer, or
-  other).
-- On performing an IPLD store get, fixed amount.
-- On performing an IPLD store put, variable on block payload size.
-- On actor creation, fixed amount.
-- On actor deletion, fixed amount.
+performed during the execution of transactions, (b) allocate chain bandwidth,
+and to (c) restrict the compute capacity of blocks so that the designated epoch
+time (30 seconds) can be sustained with contemporary hardware.
+
+Today, transactions on the Filecoin network are charged gas on these operations:
+
+- On including a chain message (variable gas units, based on length).
+- On returning a value (variable gas units, based on length of returned
+  payload).
+- On performing a call (variable gas units, depending on whether it's a simple
+  value transfer, or an invocation to actor logic).
+- On performing an IPLD store get (fixed gas units).
+- On performing an IPLD store put (variable gas units, based on block payload
+  size).
+- On actor creation (fixed gas units).
+- On actor deletion (fixed gas units).
 - On cryptographic operations:
-    - On signature verification, variable on signature type and plaintext
-      length.
-    - On hashing, fixed price.
-    - On computing the CID of an unsealed sector, variable on proof type and
-      piece characteristics.
-    - On verifying a seal, variable on the seal characteristics.
-    - On verifying a window PoSt, variable on the PoSt characteristics.
+    - On signature verification (variable gas units, depending on signature type
+      and plaintext length).
+    - On hashing (fixed gas units).
+    - On computing the CID of an unsealed sector (variable gas units, based on
+      proof type and piece properties).
+    - On verifying a seal (variable gas units, based on seal properties).
+    - On verifying a window PoSt (variable gas units, based on PoSt properties).
     - On verifying a consensus fault.
 
-None of these gas charges accounts for the costs of the execution of the actor’s
-compute logic itself. This results in an incomplete resource accounting model.
-So far, the network has coped without it because:
+None of these gas charges accounts for the actual costs of the execution of the
+actor's compute logic itself. This results in an imprecise resource accounting
+system. So far, the network has coped without it because:
 
-1. The cost of built-in actor logic is negligible when compared to the cost of
-   the above operations, most of which are triggered through syscalls.
+1. With the exception of some methods, the cost of most built-in actor logic is
+   negligible when compared to the cost of the above operations, most of which
```

Comment:
IMPORTANT: I'd say "is proportional to the cost of the above operations". As we've seen, it's not really negligible.